.github/workflows/test-all.yml

```diff
@@ -0,0 +1,29 @@
+name: Test All
+
+on:
+    push:
+        branches:
+            - main
```

Comment:
Maybe that would be an option [Running a workflow when a pull request is approved](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#running-a-workflow-when-a-pull-request-is-approved)

Reply:
Locally, you can do `pnpm test:force:all` from the root before pushing to the branch

Reply:
I think the "test affected" in the PR will not catch the situations where you have a missing dependency. (ie. an engineering error.) You don't want to do a full test for a PR because that takes too long. The point of the full build on main is to catch those dependency errors in a sort of async way because you only get notified if something is wrong.